src/main/java/cnLabs/reactivedemo/Controller/TeacherController.java

```diff
@@ -0,0 +1,79 @@
+package cnLabs.reactivedemo.Controller;
+
+import cnLabs.reactivedemo.Model.Teacher;
+import cnLabs.reactivedemo.Service.TeacherService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.stereotype.Component;
+import org.springframework.web.reactive.function.server.RouterFunction;
+import org.springframework.web.reactive.function.server.ServerResponse;
+import reactor.core.publisher.Mono;
+
+import static org.springframework.web.reactive.function.server.RequestPredicates.*;
+import static org.springframework.web.reactive.function.server.RouterFunctions.route;
+import static org.springframework.web.reactive.function.server.ServerResponse.ok;
+
+@Component
+public class TeacherController {
+
+    @Autowired
+    TeacherService teacherService;
```

Comment:
```suggestion
    private final TeacherService teacherService;
```

Reply:
Oh, should the services autowired into Controllers always be final?